File: backend/src/BiteRight.Application/Dtos/Users/ProfileDto.cs
// # ==============================================================================
// # Solution: BiteRight
// # File: ProfileDto.cs
// # Author: Łukasz Sobczak
// # Created: 12-02-2024
// # ==============================================================================

#region

using BiteRight.Application.Dtos.Currencies;
using BiteRight.Domain.Users;

#endregion

namespace BiteRight.Application.Dtos.Users;

public class ProfileDto
{
    public ProfileDto(
        CurrencyDto currency,
        string timeZoneId
    )
    {
        Currency = currency;
        TimeZoneId = timeZoneId;
    }

    public CurrencyDto Currency { get; set; }
    public string TimeZoneId { get; set; }

    public static ProfileDto FromDomain(
        Profile profile
    )
    {
        return new ProfileDto(
            CurrencyDto.FromDomain(profile.Currency),
            profile.TimeZone.Id
        );
    }
}

File: mobile/app/src/main/java/com/sobczal2/biteright/dto/users/ProfileDto.kt
package com.sobczal2.biteright.dto.users

import com.google.gson.annotations.SerializedName
import com.sobczal2.biteright.dto.currencies.CurrencyDto
import java.util.UUID

data class ProfileDto(
    @SerializedName("currency") val currency: CurrencyDto,
    @SerializedName("timeZoneId") val timeZoneId: String,
)